src/features/usage-consolidation-accordion/full-text-request-usage-table/full-text-request-usage-table.js

```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  FormattedNumber,
+  useIntl,
+} from 'react-intl';
+
+import {
+  KeyValue,
+  MultiColumnList,
+  NoValue,
+} from '@folio/stripes/components';
+
+import {
+  costPerUse as costPerUseShape,
+  platformTypes,
+} from '../../../constants';
+
+import styles from './full-text-request-usage-table.css';
+
+const monthsNames = {
+  JAN: 'jan',
+  FEB: 'feb',
+  MAR: 'mar',
+  APR: 'apr',
+  MAY: 'may',
+  JUN: 'jun',
+  JUL: 'jul',
+  AUG: 'aug',
+  SEP: 'sep',
+  OCT: 'oct',
+  NOV: 'nov',
+  DEC: 'dec',
+};
+
+const propTypes = {
+  costPerUseData: costPerUseShape.CostPerUseReduxStateShape.isRequired,
+  platformType: PropTypes.string.isRequired,
+  startMonth: PropTypes.string.isRequired,
+};
+
+const FullTextRequestUsageTable = ({
+  costPerUseData,
+  platformType,
+  startMonth,
+}) => {
+  const {
+    totals,
+    platforms,
+  } = costPerUseData?.data?.attributes?.usage;
+
+  const intl = useIntl();
+
+  const indexOfStartMonth = Object.values(monthsNames).indexOf(startMonth);
+  const months = Object.values(monthsNames).slice(indexOfStartMonth);
+
+  const monthsColumnNames = months.reduce((acc, month) => ({
+    ...acc,
+    [month]: month,
+  }), {});
+
+  const columnNames = {
+    PLATFORM: 'platform',
+    ...monthsColumnNames,
+    TOTAL: 'total',
+    PUBLISHER: 'publisher',
+  };
+
+  const [sortedColumn, setSortedColumn] = useState(columnNames.PLATFORM);
+  const [sortOrder, setSortOrder] = useState('ascending');
+
+  const platformData = platforms.map(({ name, isPublisherPlatform, counts, total }) => {
+    const countByMonth = counts.reduce((acc, value, index) => ({
+      ...acc,
+      [months[index]]: value,
+    }), {});
+
+    return {
+      [columnNames.PLATFORM]: name,
+      ...countByMonth,
+      [columnNames.TOTAL]: total,
+      [columnNames.PUBLISHER]: isPublisherPlatform,
+    };
+  });
+
+  const currentPlatforms = platformType === platformTypes.ALL
+    ? Object.values(platformTypes)
+    : [platformType];
+
+  const totalsData = currentPlatforms.map(platform => {
+    const countByMonth = totals[platform]?.counts.reduce((acc, value, index) => ({
+      ...acc,
+      [months[index]]: value,
+    }), {});
+
+    return {
+      [columnNames.PLATFORM]: platform,
+      ...countByMonth,
+      [columnNames.TOTAL]: totals[platform]?.total,
+    };
+  });
+
+  if (platformType === platformTypes.ALL) {
+    totalsData.push(totalsData[0]);
+    totalsData.shift();
+  }
+
+  const contentPlatformData = platformData
+    ? platformData.sort((a, b) => {
+      const valA = sortOrder === 'ascending' ? a : b;
+      const valB = sortOrder === 'ascending' ? b : a;
+
+      if (sortedColumn === columnNames.PLATFORM) {
+        return valA[sortedColumn].localeCompare(valB.packageName);
```

Comment:
I think you should compare to `valB[sortedColumn]`, right?

Reply:
Right! Thank you